source/extensions/transport_sockets/http_11_proxy/connect.cc

```diff
@@ -29,8 +29,21 @@ bool UpstreamHttp11ConnectSocket::isValidConnectResponse(absl::string_view respo
 
 UpstreamHttp11ConnectSocket::UpstreamHttp11ConnectSocket(
     Network::TransportSocketPtr&& transport_socket,
-    Network::TransportSocketOptionsConstSharedPtr options)
-    : PassthroughSocket(std::move(transport_socket)), options_(options) {
+    Network::TransportSocketOptionsConstSharedPtr options,
+    std::shared_ptr<const Upstream::HostDescription> host, bool legacy_behavior)
+    : PassthroughSocket(std::move(transport_socket)), options_(options),
+      legacy_behavior_(legacy_behavior) {
+  if (legacy_behavior_) {
+    legacyConstructor();
+    return;
+  }
+
+  header_buffer_.add(
+      absl::StrCat("CONNECT ", host->address()->asStringView(), " HTTP/1.1\r\n\r\n"));
```

Comment:
Hmmm, I didn't get this point. Why use the host address directly here?

PS: I am not familiar to this area. sorry if this is a stupid quesion.

Reply:
I'm not sure what else I could use to get the host address. Did you have another idea?